lib/note-content-editor.tsx

```diff
@@ -167,6 +173,86 @@ class NoteContentEditor extends Component<Props> {
     }
   };
 
+  completionProvider: (
+    selectedNoteId: T.EntityId | null
+  ) => languages.CompletionItemProvider = (selectedNoteId) => {
+    return {
+      triggerCharacters: ['['],
+
+      provideCompletionItems(model, position, context, token) {
+        const line = model.getLineContent(position.lineNumber);
+        if (!line.includes('[')) {
+          return null;
+        }
```

Comment:
we're getting the `precedingOpener` right below. we can probably avoid this check and avoid scanning the line twice.

```ts
const precedingOpener = line.lastIndexOf(…);
if (-1 === precedeingOpener) {
	return null;
}
```